CRM/Core/Payment.php

```diff
@@ -999,9 +999,24 @@ protected function getCurrency($params) {
   }
 
   /**
-   * Get the currency for the transaction.
+   * Get the amount to be processed, rounded to the proscribed decimal places.
    *
-   * Handle any inconsistency about how it is passed in here.
+   * The amount value should always be 'clean' (using dot separator) by the time it reaches this
+   * function as we are cleaning close to point of submit, so only rounding takes place here.
+   *
+   * @param array $params
+   * @param int $decimalPlaces
+   * @return float
+   */
+  protected function getAmountDotDecimalSeparator($params, $decimalPlaces = 2) {
```

Comment:
I feel like this kind of function would be better in CRM_Utils_Money as we already have a couple of formatters in there and a function to get decimal places for currency.  It would tie in nicely with my work here on providing different money formatters: https://github.com/civicrm/civicrm-core/pull/10641/files#diff-b30591c63856146cff020415a7df8f42L41

Reply:
I don't mind wrapping a function there but I feel the pattern for this class is to get people to stop accessing $params & move towards using get functions from the child classes

Reply:
@mattwire I feel nervous about changing an existing function - I'd rather add something new that is really specific & clear - we could deprecate getAmount

Reply:
@eileenmcnaughton How about extending getAmount with an optional parameter, which is a constant defined in CRM_Utils_Money to specify what type of amount should be returned.  It would default to what it's doing now, but we could easily override without re-implementing for each payment processor just by passing a suitable constant in?

Reply:
I don't think that many processors use getAmount at the moment & I'm not 100% sure what it's behaviour IS at the moment - my take is it probably will format to 1.000,77 if the site thousand separator is a dot.

We ALSO don't know what format the gateways want to receive the amount value in - that is gateway specific. So, I'm very nervous about changing any other gateway.

if we add a new parameter to the signature then we are getting into the situation where it is basically doing totally different things depending on what is passed in - which is just confusing. 

In the past $params['amount'] has sometimes not been the amount key (eg. $params['total_amount'] in some flows. That should be OK-ish now but having a function to retrieve it means we can handled that if it changes.

Also, in the past it in not 100% clear to me that the format was always 1000.56000 or 1000.56 - but that should always be the case now since we are cleaning money close to submission now. It occurred to me that we can test if by using the alterPaymentProcessor hook to check the params passed through - which would be a good testing extension. 

I DO think ideally getAmount would be like the similar Omnipay function - ie

Reply:
Bugger - if this is a payflow bug not paypal I'm just gonna close this & merge @yashodha's PR - PayflowPro is totally obscure & I'm surprised it even still works. I doubt any Euro sites would use it


FWIW I think to get something I really felt happy with I would want to have an Amount object & then use 
```
$this->getAmountObj()->getAmountInCents() ;
$this->getAmountObj()->getAmountLocaleFormattedLong() ;
$this->getAmountObj()->getAmountDotDecimalFormatted() ;
```

etc

Reply:
Hmm - but if we did that we would be defining a new interface so I would want to wrap those functions on the CRM_Core_Payment class rather than have them call it direct